beamformer/test/prebeamform_reorder_test.py

```diff
@@ -0,0 +1,214 @@
+        The number of frequency channels out of the FFT.
+        NB: This is not the number of FFT channels per stream.
+        The number of channels per stream is calculated from this value.
+    num_samples_per_channel: int
+        The number of time samples per frequency channel.
+    n_times_per_block: int
+        Number of blocks to break the number of samples per channel into.
+    """
+    # Now to create the actual PreBeamformReorderTemplate
+    # 1. Array parameters
+    # - Will be {ants, chans, samples_per_chan, batches}
+    # - Will pass num_{ants, samples_per_channel} parameters straight into Template instantiation
+
+    # This integer division is so that when num_ants % num_channels !=0 then the remainder will be dropped.
+    # - This will only occur in the MeerKAT Extension correlator.
+    # TODO: Need to consider the case where we round up as some X-Engines will need to do this to capture all the channels.
+    n_channels_per_stream = num_channels // num_ants // 4
+
+    pol = 2  # Always
+
+    # 2. Initialise GPU kernels and buffers.
+    ctx = accel.create_some_context(device_filter=lambda x: x.is_cuda, interactive=False)
+    queue = ctx.create_command_queue()
+
+    template = PreBeamformReorderTemplate(
+        ctx,
+        n_ants=num_ants,
+        n_channels=n_channels_per_stream,
+        n_samples_per_channel=num_samples_per_channel,
+        n_batches=batches,
+    )
+    preBeamformReorder = template.instantiate(queue)
+    preBeamformReorder.ensure_all_bound()
+
+    bufSamples_device = preBeamformReorder.buffer("inSamples")
+    bufSamples_host = bufSamples_device.empty_like()
+
+    bufReordered_device = preBeamformReorder.buffer("outReordered")
+    bufReordered_host = bufReordered_device.empty_like()
+
+    # 3. Generate random input data - need to modify the dtype and shape of the array as numpy does not have a packet
+    # 8-bit int complex type.
+
+    # Get the shape in place because we used Dimension objects in the template
+    inputDataShape = (
+        batches,
+        num_ants,
+        n_channels_per_stream,
+        num_samples_per_channel,
+        pol,
+    )
+
+    # DEBUG: Inject sample to trace
+    # bufSamples_host.dtype = np.uint8
+    # bufSamples_host[:] = np.zeros(bufSamplesInt8Shape, dtype=np.uint8)
+    # bufSamples_host[0][0][0][0][0] = 1 #Pol0 Sample0 Real
+    # bufSamples_host[0][0][0][0][1] = 55 #Pol0 Sample0 Imag
+
+    # bufSamples_host[0][0][0][1][0] = 77 #Pol0 Sample1 Real
+    # bufSamples_host[0][0][0][1][1] = 33 #Pol0 Sample1 Imag
+
+    # OR
+
+    # Inject random data for test.
+    bufSamples_host[:] = np.random.randint(
+        low=1,
+        high=pow(2, 16) - 2,
+        size=inputDataShape,
+        dtype=bufSamples_host.dtype,
+    )
+
+    # Convert back to uint16 as to deal with the complex (real and imag) as single word pairs.
+    bufSamples_host.dtype = np.uint16
+    bufSamples_host = np.ndarray.view(bufSamples_host, dtype=np.uint16)
```

Comment:
The second line is the proper way to do what you are doing in the first line.
```suggestion
    # Convert back to uint16 as to deal with the complex (real and imag) as single word pairs.
    bufSamples_host = np.ndarray.view(bufSamples_host, dtype=np.uint16)
```

In general, don't alter the dtype directly. It's bad practice. But if you use my suggestion from the precorrelation reorder unit test, you won't even need to do this because it'll be in the right format already.

Reply:
Removed.